File: Event.cpp
#include "Event.h"

namespace xelous
{
	const bool Event::operator==(const EventId& id) const
	{
		return Id == id;
	}

	const bool Event::operator==(const Event& other) const
	{
		return other.Id == Id;
	}

	Event::Event(const Event& other)
		:
		Id(other.Id)
	{

	}
	
	Event::Event(const EventId& id)
		:
		Id(id)
	{
	}
}

File: Tool/SMGen/Event.cs
using System;

namespace xelous
{
    public class Event
    {
        private string mName;

        public Event(string pName)
        {
            mName = pName;
        }

        public string Name
        {
            get { return mName; }
            set { mName = value; }
        }
    }
}